### PR TITLE
add exception handling in getKeys

### LINKED ISSUE #329: Get @ barbara's atsign working!
@nickelskevin or @sarika01 or @tinashe404  which ticket is this connected to, so we can mark here?  TY!

### PR DESCRIPTION
**Pull request by sitaram-kalluri**
**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: at_client/lib/src/client/at_client_impl.dart [hunk 1]
```diff
@@ -470,7 +470,11 @@ class AtClientImpl implements AtClient {
     var result = <AtKey>[];
     if (getKeysResult.isNotEmpty) {
       getKeysResult.forEach((key) {
-        result.add(AtKey.fromString(key));
+        try {
+          result.add(AtKey.fromString(key));
+        } on InvalidSyntaxException {
```
Comment (VJag) [inline at_client/lib/src/client/at_client_impl.dart:475]: Better to handle a broader exception here

Comment (VJag) [reply at_client/lib/src/client/at_client_impl.dart:475]: I mean an additional catch block for generic exceptions with message 

_logger.severe('Unable to form key $key');

Comment (sitaram-kalluri) [reply at_client/lib/src/client/at_client_impl.dart:475]: Sure Jagan.

File: at_client/lib/src/client/at_client_impl.dart [hunk 2]
```diff
@@ -470,7 +470,13 @@ class AtClientImpl implements AtClient {
...
+          _logger.severe('Failed to return. $key is not a well-formed key');
```
Comment (VJag) [inline at_client/lib/src/client/at_client_impl.dart]: Failed to return. Please remove that text.

Comment (VJag) [reply at_client/lib/src/client/at_client_impl.dart]: Also log the exception e

**Review by VJag — APPROVED** [review]

**Pull request merged**